Finmer.Game/Gameplay/Scripting/ContentScriptLibrary.cs

```diff
@@ -0,0 +1,81 @@
+﻿/*
+ * FINMER - Interactive Text Adventure
+ * Copyright (C) 2019-2024 Nuntis the Wolf.
+ *
+ * Licensed under the GNU General Public License v3.0 (GPL3). See LICENSE.md for details.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+using System;
+using System.Linq;
+using static Finmer.Gameplay.Scripting.LuaApi;
+
+namespace Finmer.Gameplay.Scripting
+{
+
+    /// <summary>
+    /// Script library containing functions related to the content repository, mainly for inspecting modules and assets.
+    /// </summary>
+    internal static class ContentScriptLibrary
+    {
+
+        /// <summary>
+        /// Register the library contents in the ScriptContext.
+        /// </summary>
+        /// <param name="context"></param>
+        public static void Inject(ScriptContext context)
+        {
+            IntPtr state = context.LuaState;
+
+            // Modules table
+            lua_createtable(state, 0, 4);
+            context.RegisterFunction("IsModuleLoadedByID", ExportedContentIsModuleLoadedByID);
+            context.RegisterFunction("IsModuleLoadedByName", ExportedContentIsModuleLoadedByName);
+            context.RegisterFunction("IsAssetLoadedByID", ExportedContentIsAssetLoadedByID);
+            context.RegisterFunction("IsAssetLoadedByName", ExportedContentIsAssetLoadedByName);
+            lua_setglobal(state, "Content");
+        }
+
+        private static int ExportedContentIsModuleLoadedByID(IntPtr state)
+        {
+            var string_id = luaL_checkstring(state, 1);
+            if (!Guid.TryParse(string_id, out var id))
+                return luaL_argerror(state, 1, "invalid id format");
+
+            lua_pushboolean(state, GameController.LoadedModules.Any(metadata => id == metadata.ID));
+
+            return 1;
+        }
+
+        private static int ExportedContentIsModuleLoadedByName(IntPtr state)
+        {
+            var name = luaL_checkstring(state, 1);
+
+            lua_pushboolean(state, GameController.LoadedModules.Any(metadata => name == metadata.Title));
```

Comment:
I think there's probably an argument to be made that someone who's used to the case-insensitivity of save data would be somewhat surprised to find that these are case-sensitive. Although save data might be the outlier here -- I think all the "constructors" etc. use case-sensitive asset names, so perhaps it would be the other way around. (Also, I did go and test upper-case `UUID`s, and that worked, so those are case-insensitive. I also tested asset names, which are case-sensitive like I expected.)

Reply:
I agree, asset constructors are also case-sensitive so by and large these functions should be expected to work the same.